### PR TITLE
NO-ISSUE: Check events in kubeapi subsystem

### PR DESCRIPTION
**Pull request by rollandf**
# Assisted Pull Request

## Description
Verify that the event URL in the CRDs is actually bringing events.

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [x] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @nmagnezi 
/cc @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: subsystem/kubeapi_test.go [hunk 1]
```diff
@@ -1620,6 +1622,11 @@ var _ = Describe("[kube-api]cluster installation", func() {
 			return infraEnv.Status.InfraEnvDebugInfo.EventsURL
 		}, "30s", "10s").Should(MatchRegexp(fmt.Sprintf("/v2/events.*infra_env_id=%s", infraEnv.ID.String())))
 
+		infraEnvCr := getInfraEnvCRD(ctx, kubeClient, infraEnvKey)
+		events, err := testEventUrl(infraEnvCr.Status.InfraEnvDebugInfo.EventsURL)
+		Expect(err).To(BeNil())
+		Expect(events).NotTo(BeEmpty())
```
Comment (nmagnezi) [inline subsystem/kubeapi_test.go:1628]: I haven't cherry-picked this yet, but is it possible to check the Payload len so we can know we have events count > 0?

**Review by nmagnezi — COMMENTED** [review]
Thanks for adding this. One question in the comments.

File: subsystem/kubeapi_test.go (hunk 1 above)
Comment (rollandf) [reply subsystem/kubeapi_test.go:1628]: Will this always be true in all tests? If yes, we can this check in the `testEventUrl` func.

**Comment by rollandf** [comment]
/hold
Checking if this test pass, then I will add to other tests also.

File: subsystem/kubeapi_test.go (hunk 1 above)
Comment (rollandf) [reply subsystem/kubeapi_test.go]: Added the check in the func. Let see if tests pass

**Comment by rollandf** [comment]
/hold cancel
@nmagnezi PTAL

**Comment by nmagnezi** [comment]
I tested this with printing the `list` events in `testEventUrl()`

```go
time="2021-12-30T13:06:27Z" level=info msg="DEBUG: event: &{Category: ClusterID:1966d466-f901-4878-8ed1-14cdad9f45a0 EventTime:2021-12-30T13:06:24.511Z HostID:<nil> InfraEnvID:<nil> Message:0xc000d4ab20 Props: RequestID: Severity:0xc000d4ab30}"
time="2021-12-30T13:06:27Z" level=info msg="DEBUG: event: &{Category: ClusterID:1966d466-f901-4878-8ed1-14cdad9f45a0 EventTime:2021-12-30T13:06:27.024Z HostID:7c674c5e-3575-4836-998a-1e78e9f3914f InfraEnvID:d831ef3e-cd3c-49f5-bf4e-1d90135f18e1 Message:0xc000d4abb0 Props: RequestID: Severity:0xc000d4abc0}"
time="2021-12-30T13:06:27Z" level=info msg="DEBUG: event: &{Category: ClusterID:1966d466-f901-4878-8ed1-14cdad9f45a0 EventTime:2021-12-30T13:06:27.041Z HostID:7c674c5e-3575-4836-998a-1e78e9f3914f InfraEnvID:d831ef3e-cd3c-49f5-bf4e-1d90135f18e1 Message:0xc000d4ac40 Props: RequestID: Severity:0xc000d4ac50}"
```

looks as expected.

**Review by nmagnezi — APPROVED** [review]
/lgtm

**Pull request merged**